packages/components/src/ActionCell/useLogic/useLogic.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  type MouseEventHandler,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import { type ActionsCellProps } from '../ActionCell';
+import { type NestedAction, type SingleAction } from '../types';
+
+type UseLogicParams<T> = ActionsCellProps<T>;
+
+export const useLogic = <T>({ actions, row }: UseLogicParams<T>) => {
+  const [isBlockingAction, setIsBlockingAction] = useState(false);
+  const { main, secondary } = actions;
+  const isLoading = main.some((action) => {
```

Comment:
```suggestion
  const isLoading = main.some((action) => action?.loading
```

Этого должно быть достаточно